### PR TITLE
ci(*:skip) Add Python version updater

### PR DESCRIPTION
**Pull request by charlesbvll**
### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: dev/update_python.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            # Python files
+            "**/*.py": [
+                # Update version assignments
+                (
+                    r'(["\'])' + version_pattern + r'(["\'])',
+                    r"\g<1>" + new_full_version + r"\g<2>",
+                ),
+            ],
+            # Documentation files
+            "doc/source/conf.py": [
```
Comment (Robert-Steiner) [inline dev/update_python.py:60]: The doc files don't currently seem to use the full version.

**Review by Robert-Steiner — COMMENTED** [review]
It appears that the script can only be executed by running `python dev/update_python.py`. Is it possible to call it from within the `dev` directory as well?

File: doc/source/conf.py [hunk 2]
```diff
@@ -98,6 +98,8 @@
 .. |ubuntu_version| replace:: 22.04
 .. |setuptools_version| replace:: 70.3.0
 .. |pip_version| replace:: 24.1.2
+.. |python_version| replace:: 3.10
+.. |python_full_version| replace:: 3.10
```
[the remark this answers is not in the document]
Comment (charlesbvll) [reply doc/source/conf.py:102]: Whoops I seem to have committed some of my testing ^^ I will fix that

[the remark this answers is not in the document]
Comment (charlesbvll) [reply doc/source/conf.py]: @Robert-Steiner Solved!

File: dev/update_python.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+                    + r"/>`_)",
+                    r"`Python "
+                    + new_full_version
+                    + " <https://docs.python.org/"
+                    + new_major_minor
+                    + "/>`_",
+                ),
+            ],
+            # PO files for localization
+            "doc/locales/*/LC_MESSAGES/framework-docs.po": [
```
Comment (Robert-Steiner) [inline dev/update_python.py:85]: Those currently don't use the full version format, only `major.minor`. When I run it with `--patch-only` I get:

```diff
- Flower requires at least `Python 3.9 <https://docs.python.org/3.9/>`_, but `Python 3.10 <https://docs.python.org/3.10/>`_ or above is recommended.
+ Flower requires at least `Python 3.9.23 <https://docs.python.org/3.9/>`_, but `Python 3.10 <https://docs.python.org/3.10/>`_ or above is recommended.
```

Should we move them to the `else` branch or do we want to use the full version format?

Comment (charlesbvll) [reply dev/update_python.py:85]: Yes they should indeed be removed, I'll push the fix

**Review by Robert-Steiner — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**